### DISCUSSION TITLE
Overriding uno's implementation of a control for a platform

### DISCUSSION THREAD

**Discussion by cmonto** [unoplatform/uno#6481] [Q&A] [answered]
I was wondering if it is possible to override uno's implementation of a control from my class library - much like how Xamarin form's renderer and registration works from any class library. I may want to keep uno's implementation of a control (say ScrollViewer) for one platform (say for iOS) and override with my implementation for another (say for Android).  I know uno does not have a renderer Architecture like Xamarin Forms, so any way to achieve this? 

Thanks in advance.

**Comment by jeromelaban** [comment, marked as answer]
You can use [conditional XAML, or conditional C#](https://platform.uno/docs/articles/platform-specific-csharp.html?q=conditional) to do just that. You can also use styling for `Control` based implementations, and have a style specific per platform and have it defined by default with conditional XAML.

**Comment by sasakrsmanovic** [comment]
Hello @cmonto ; this may not be exactly what you are looking for but could help. https://platform.uno/blog/hosting-native-controls-moving-from-xamarin-forms-renderers-to-uno-platform-controls/ 

BTW, I'd be very curious to learn more about your project and what you are working on. I'm at sasha at platform dot uno if you'd like to reach out to chat at any time.

**Comment by sasakrsmanovic** [comment]
@cmonto Also, if Jerome's answer was good, it would be great to mark the question as answered; it helps community know what the good answers are.

**Comment by cmonto** [comment]
Hi @sasakrsmanovic.. I appreciate both answers though not exact. I am trying to port one of my WP project to Uno. Started with Xamarin Forms and now have taken a step back.
